### PR TITLE
ref: validate userfeedback id

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -187,6 +187,10 @@ private SentryEvent processEvent(
   public void captureUserFeedback(UserFeedback userFeedback) {
     Objects.requireNonNull(userFeedback, "SentryEvent is required.");
 
+    if (userFeedback.getEventId() == null || userFeedback.getEventId() == SentryId.EMPTY_ID) {
```
Comment (maciejwalkowiak) [inline sentry/src/main/java/io/sentry/SentryClient.java:190]: As far as I can see `userFeedback#eventId` can never be `null`, thus should be marked as `@NotNull` and this part of if statement wouldn't be needed.

Comment (bruno-garcia) [reply sentry/src/main/java/io/sentry/SentryClient.java:190]: doesn't this assume tooling respects the annotations? aren't those just hints that tooling must support? Or it's guaranteed compiler won't let this compile?

Comment (marandaneto) [reply sentry/src/main/java/io/sentry/SentryClient.java]: those annotations are just a convention adopted by IntelliJ based IDEs, so it still compiles (for Java-even with warnings), but for Kotlin, it doesn't compile iirc.
I think its fine to keep this `if` for the sake of safety, but annotating as @maciejwalkowiak said is also helpful

Comment (maciejwalkowiak) [reply sentry/src/main/java/io/sentry/SentryClient.java]: Not respecting this annotation should result in warning that we must fix. Not sure if currently the build fails but I believe it should.

**Review by maciejwalkowiak — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,8 @@
 * feat: Manually capturing User Feedback
 * Enhancement: Set environment to "production" by default.
 * Enhancement: Make public the Breadcrumb constructor that accepts a Date #1012
-
+* ref: Validate event id on user feedback submission
```
Comment (philipphofmann) [inline CHANGELOG.md:6]: Thanks for this. I should also add this to Cocoa.

**Pull request merged**